### PR TITLE
feat: swaps the underlying curl lib for guzzle

### PR DESCRIPTION
**Pull request by Justintime50**
# Description

- Adopts `Guzzle` as the HTTP client for this library. This should provide a much more consistent experience, better encoding, faster request times in some cases. By using Guzzle we can remove the old SSL logic and completely rip out the curl logic we homegrew.

# Testing

- [x] Fixed various tests with new encoding, new rates, etc
- [x] E2E tested timeouts, failures to connect without a connection, etc
- [x] Re-recorded all cassettes to use the new HTTP client and catch various edge cases

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

**Review by nwithan8 — APPROVED** [review]

**Comment by Justintime50** [comment]
Drafting till I can merge the other open PR then rebase this one

File: lib/EasyPost/EasyPost.php [hunk 1]
```diff
@@ -118,50 +109,26 @@ public static function setApiVersion($apiVersion)
...
-    }
-
-    /**
-     * Set time in milliseconds to wait for a connection.
-     *
-     * Zero or null means no timeout.
-     *
-     * @param int|null $connectTimeout
-     */
-    public static function setConnectTimeout($connectTimeout)
```
Comment (jchen293) [inline lib/EasyPost/EasyPost.php:139]: Why is `setConnectTimeout` removed? I assume Guzzle doesn't support this?

**Review by jchen293 — APPROVED** [review]

File: lib/EasyPost/EasyPost.php (hunk 1 above)
Comment (Justintime50) [reply lib/EasyPost/EasyPost.php:139]: Correct, they only have one timeout param instead of allowing for request and connection.
